### PR TITLE
handbrake: 1.6.1 -> 1.7.3

### LINKED ISSUE #268104: Update request: handbrake 1.6.1 → 1.7.3
- Package name: handbrake
- Latest released version: 1.7.3
- Current version on the unstable channel: 1.6.1
- Current version on the stable/release channel: 1.6.1
- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@Anton-Latukha  
@wmertens

https://github.com/HandBrake/HandBrake/releases/tag/1.7.3

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by uninsane**
## Description of changes

- fixes existing build break
- upstream changelogs: <https://github.com/HandBrake/HandBrake/releases>
  - `gtk` component is ported from autotools to meson
  - upstream mentions a new library, `libdovi`: i haven't acted on this; seems to be optional
- tested on x86_64-linux:
  - `nix build '.#handbrake.tests.basic-conversion'`
  - checked that both the gui (`'.#handbrake'`) and headless (`useGtk = false`) versions build
  - transcoded a short H.264 video through the GUI
  - did not test darwin build

Closes #268104.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/handbrake/default.nix [hunk 1]
```diff
@@ -86,64 +90,70 @@
...
+  # https://github.com/HandBrake/HandBrake/blob/master/contrib/ffmpeg/module.defs
+  ffmpeg-version = "6.1";
+  ffmpeg-hb = ffmpeg-full.overrideAttrs (old: {
     version = ffmpeg-version;
     src = fetchurl {
       url = "https://www.ffmpeg.org/releases/ffmpeg-${ffmpeg-version}.tar.bz2";
-      hash = "sha256-OaC8yNmFSfFsVwYkZ4JGpqxzbAZs69tAn5UC6RWyLys=";
+      hash = "sha256-632j3n3TzkiplGq0R6c0a9EaOoXm77jyws5jfn9UdhE=";
     };
     patches = old.patches or [ ] ++ [
```
Comment (antonmosich) [inline pkgs/applications/video/handbrake/default.nix:114]: Maybe also add parentheses here, to make it easier to read.
```suggestion
    patches = (old.patches or [ ]) ++ [
```

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 297984` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>handbrake</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Looks like darwin requires `xcrun` added to nativeBuildInputs.

**Comment by wegank** [comment]
If `metal` is required for the build, this means that upstream now has Xcode as a non-free dependency, so the package should be marked as broken on darwin.

File: pkgs/applications/video/handbrake/default.nix [hunk 2]
```diff
@@ -26,7 +26,7 @@
 , numactl
 , writeText
   # Processing, video codecs, containers
-, ffmpeg_5-full
+, ffmpeg-full
```
Comment (SuperSandro2000) [inline pkgs/applications/video/handbrake/default.nix:29]: Isn't the default standard ffmpeg not enough? And if so why?

File: pkgs/applications/video/handbrake/default.nix [hunk 3]
```diff
@@ -286,6 +313,10 @@ let
       version = testers.testVersion { package = self; command = "HandBrakeCLI --version"; };
     };
 
+    # for convenience
+    passthru.ffmpeg-hb = ffmpeg-hb;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/handbrake/default.nix:317]: We can reduce code duplication here by using inherit

File: pkgs/applications/video/handbrake/default.nix (hunk 2 above)
Comment (bachp) [reply pkgs/applications/video/handbrake/default.nix:29]: See https://github.com/NixOS/nixpkgs/issues/271863 for context

**Pull request merged**